### ISSUE TITLE
Should benches be run when testing

### ISSUE THREAD

**Issue by BusyJay** [rust-lang/cargo#2984]
Benches under src directory will be run as test when execute command

```
cargo test
```

But the benches under benches directory will not be automatically run as test. I think these code should be run as test too. Benches are not run frequently and may be broken after some changes.

**Comment by jonas-schievink** [comment]
This might potentially break CI for projects testing on stable/beta, since the benches will only compile on nightly.

**Comment by alexcrichton** [comment]
Yeah currently the benchmarking infrastructure is all nightly-only, unfortunately, so I'd want to wait until that hit stable before we turn this on. Once it's stable, though, this sounds like a great idea!

**Comment by Robbepop** [comment]
**I want to argue against the proposal behind this issue!**

My current project suffers from the fact, that benches are run alongside with tests while executing "cargo test" since my benchmarks are even without multiple iterations very compute intense and thus very slow. My idea is to no longer run any benches during "cargo test" but to make it possible to mark a benchmark function as #[test] besides its #[bench] annotation to make it run with "cargo test" as it is today to prevent the scenario of outdated test that aren't too slow to be executed all the time.

**TL;DR:**
 - do no longer run any `#[bench]` during `cargo test` because many `#[bench]` functions are very compute intense.
 - simply give `#[bench]` functions another `#[ŧest]` attribute to make them run on `cargo test` to prevent them from getting outdated.

**Comment by carols10cents** [comment]
Because it sounds like the answer to the question in the title of this issue is "no", I'm going to close this issue. Thanks!
